File: version_Java/sample_1_di/src/main/java/com/example/sample_1_di/common/argument/ScannerArgumentResolver.java
package com.example.sample_1_di.common.argument;


import org.springframework.stereotype.Component;

import java.io.InputStream;
import java.util.Scanner;

//@Component
public class ScannerArgumentResolver implements ArgumentResolver {
    @Override
    public Argument resolve(InputStream stream) {
        Scanner scanner = new Scanner(stream);

        int a = scanner.nextInt();
        int b = scanner.nextInt();

        return new Argument(a, b);
    }
}

File: version_Kotlin/sample_2_di/src/main/kotlin/com/example/sample_2_di/argument/ScannerArgumentResolver.kt
package com.example.sample_2_di.argument

import org.springframework.stereotype.Component
import java.io.InputStream
import java.util.*

@Component
class ScannerArgumentResolver: ArgumentResolver{

    override fun resolve(stream:InputStream): Argument {
        val scanner = Scanner(stream)

        val a = scanner.nextInt()
        val b = scanner.nextInt()

        return Argument(a, b)

    }
}

interface ArgumentResolver {
    fun resolve(stream : InputStream): Argument
}

data class Argument(val a:Int, val b:Int)